generator/integration_tests/golden/v1/internal/golden_kitchen_sink_rest_metadata_decorator.cc

```diff
@@ -161,6 +165,7 @@ GoldenKitchenSinkRestMetadata::ExplicitRouting2(
 void GoldenKitchenSinkRestMetadata::SetMetadata(
       rest_internal::RestContext& rest_context,
       Options const& options, std::vector<std::string> const& params) {
+  rest_context.AddHeader("x-goog-api-version", kServiceApiVersion);
```

Comment:
Consider passing `kServiceApiVersion as an argument (maybe `absl::optional<std::string>`? or `absl::string_view` and `.empty()` to imply "not set"?)

Having one function set all the system-level headers makes testing easier, the fixes (if any) do not involve the generated.  And future changes (e.g.  "This option let's you override the api version") can be tested and implemented without changes to the generator or generator code.